.github/workflows/python_compile_requirements.yml

```diff
@@ -0,0 +1,33 @@
+name: Compile python requirements
+on:
+  schedule: 
+    - cron: 00 00 * * *
+  workflow_dispatch:
+
+jobs:
+  compile_requirements:
+    name: Compile requirements and commit changes
+    runs-on: ubuntu-22.04
+    container:
+      image: python:3.11-slim
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: Install latest gdal to pin python version, setup git
+        run: |
+          export DEBIAN_FRONTEND=noninteractive
+          apt-get update && apt-get -y install libgdal-dev git
```

Comment:
This was sort of interesting. I wanted to use a debian base image so that we could get latest libgdal-dev, which by default is not available on ubuntu (could only get 3.4.1 instead of 3.6.2, even with updates). Figured I'd use this python one for consistency. However, it doesn't have git installed. We need git for both compiling requirements (since some sources are from git repos - geosupport and my fork of sqlfluff) as well as the action for opening a PR.

Checkout doesn't need git installed... but if it isn't, it uses github's rest api, which doesn't actually make the checked out repo a git repo! It just copies the files. So the action to create the PR was failing due to it not actually being a git repo.

So solution is to install git before running checkout action. Kinda interesting!